runtime/dart_isolate.cc

```diff
@@ -251,23 +251,43 @@ std::weak_ptr<DartIsolate> DartIsolate::CreateRootIsolate(
   DartErrorString error;
   Dart_Isolate vm_isolate = nullptr;
   auto isolate_flags = flags.Get();
-  /// TODO(b/72025) This will be where we call Dart_CreateIsolateInGroup if
-  /// spawning_isolate != nullptr.
-  vm_isolate = CreateDartIsolateGroup(
-      std::move(isolate_group_data), std::move(isolate_data), &isolate_flags,
-      error.error(),
-      [](std::shared_ptr<DartIsolateGroupData>* isolate_group_data,
-         std::shared_ptr<DartIsolate>* isolate_data, Dart_IsolateFlags* flags,
-         char** error) {
-        return Dart_CreateIsolateGroup(
-            (*isolate_group_data)->GetAdvisoryScriptURI().c_str(),
-            (*isolate_group_data)->GetAdvisoryScriptEntrypoint().c_str(),
-            (*isolate_group_data)->GetIsolateSnapshot()->GetDataMapping(),
-            (*isolate_group_data)
-                ->GetIsolateSnapshot()
-                ->GetInstructionsMapping(),
-            flags, isolate_group_data, isolate_data, error);
-      });
+
+  IsolateMaker isolate_maker;
+#if FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_PROFILE || \
+    FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_RELEASE
+  if (spawning_isolate) {
+    isolate_maker =  [spawning_isolate](
+            std::shared_ptr<DartIsolateGroupData>* isolate_group_data,
+            std::shared_ptr<DartIsolate>* isolate_data,
+            Dart_IsolateFlags* flags, char** error) {
+      return Dart_CreateIsolateInGroup(
+          /*group_member=*/spawning_isolate->isolate(),
+          /*name=*/"spawn",
+          /*shutdown_callback=*/nullptr,
+          /*cleanup_callback=*/nullptr,
+          /*child_isolate_data=*/isolate_data,
+          /*error=*/error);
+        };
+  }
+#endif
+
+  if (!isolate_maker) {
+    isolate_maker = [](std::shared_ptr<DartIsolateGroupData>*
+                           isolate_group_data,
+                       std::shared_ptr<DartIsolate>* isolate_data,
+                       Dart_IsolateFlags* flags, char** error) {
+      return Dart_CreateIsolateGroup(
+          (*isolate_group_data)->GetAdvisoryScriptURI().c_str(),
+          (*isolate_group_data)->GetAdvisoryScriptEntrypoint().c_str(),
+          (*isolate_group_data)->GetIsolateSnapshot()->GetDataMapping(),
+          (*isolate_group_data)->GetIsolateSnapshot()->GetInstructionsMapping(),
+          flags, isolate_group_data, isolate_data, error);
```

Comment:
In case of error, the error message must be printed to the log (like elsewhere in this TU). Also, can you make sure we don't have to free the error message buffer? The Dart APIs are a bit inconsistent about what to do with the error buffer. Something you have to `free` it otherwise its a small leak.

Reply:
The tonic class `DartErrorString` is taking care of that here, it was a bit harder to follow before.  It should be more clear now.